### DISCUSSION TITLE
TypeError: Cannot read property 'join' of undefined when use HOC component

### DISCUSSION THREAD

**Discussion by namlq93** [vercel/next.js#17397] [Help] [answered]
I have a HOC component: 

```
export const withAuth = Component => () => {
    // get user info from context here ...

    return props => {
        if (userInfo.role == 2) {
            return (
                <>
                    <Component {...props} />
                </>
            );
        } else {
            Router.push('/login');
        }
    };
};
```

And I wrap page by HOC component: 

```
import { withAuth } from '~/utils/withPrivateRoute';

const AdminPublishedPosts: NextPage = () => {
    return (
        <AdminLayout>
            <AdminPublishedPostsContainer />
        </AdminLayout>
    );
};
AdminPublishedPosts.getInitialProps = async (ctx: NextPageContext) => {
    serverSideAuthHandler(ctx);

    return {
        title: 'Title',
    };
};

export default withAuth(AdminPublishedPosts);


```

But I've got an error: ` TypeError: Cannot read property 'join' of undefined`

![next2](https://user-images.githubusercontent.com/41458989/94372024-c033c200-0124-11eb-9867-9821c1e264ed.PNG)

How can I fix it? Thank you so much!

**Comment by jamesmosier** [comment, marked as answer]
You must return something from your function components, _always_. So you could try:

```js
Router.push('/login');
return null;
```

**Reply by namlq93** [reply]
> You must return something from your function components, _always_. So you could try:
> 
> ```js
> Router.push('/login');
> return null;
> ```

I've tried but it still gets the error.

**Reply by jamesmosier** [reply]
What version of Next are you running? Have you tried the latest stable version (9.5.3)?

Have you tried returning `return <div/>` just to ensure the issue exists where you believe it does?

**Reply by namlq93** [reply]
@jamesmosier I'm using next 9.5.3 
I've tried `return <div/>`, but it still gets the error :(

**Reply by jamesmosier** [reply]
Do you have a repository you could share?

**Reply by jamesmosier** [reply]
I think this might be related to https://github.com/vercel/next.js/pull/17430

Could you update to `9.5.4-canary.23` and see if the issue still persists?

**Reply by jonrrivera** [reply]
I had the same issue and it worked for me using `9.5.4-canary.23`. It initially stopped working using `latest`
